code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -844,19 +846,16 @@
 	if (!istype(user))
 		return
 
-	if(istype(target,/obj/critter)) //gods how I hate /obj/critter
-		if(user.a_intent == INTENT_DISARM)
-			src.disarm(target,user)
+	if(ismob(target) || iscritter(target)) //gods how I hate /obj/critter
+		if (!isflockmob(target))
+			src.try_cage(target, user)
 			return
 
 	if(user.a_intent == INTENT_HARM)
 		if(HAS_ATOM_PROPERTY(target,PROP_ATOM_FLOCK_THING))
 			if(isflockdeconimmune(target))
 				return
 			actions.start(new /datum/action/bar/flock_decon(target), user)
-		else
-			..()
-		return
```

Comment:
I think this was a part of early return? Like if you are deconstructing you won't be doing anything else

Reply:
Well we don't want to call parent, since that just does an attack action, and early returning would prevent the conversion action on harm intent which seems unintuitive since it works on all other intents?